packages/node/src/module.ts

```diff
@@ -1,45 +1,63 @@
-import { basename, dirname } from '@sentry/utils';
+import { posix, sep } from 'path';
+
+const isWindowsPlatform = sep === '\\';
 
 /** normalizes Windows paths */
-function normalizePath(path: string): string {
+function normalizeWindowsPath(path: string): string {
   return path
     .replace(/^[A-Z]:/, '') // remove Windows-style prefix
     .replace(/\\/g, '/'); // replace all `\` instances with `/`
 }
 
 /** Gets the module from a filename */
-export function getModule(filename: string | undefined): string | undefined {
+export function getModule(
+  filename: string | undefined,
+  normalizeWindowsPathSeparator: boolean = isWindowsPlatform,
+): string | undefined {
   if (!filename) {
     return;
   }
 
-  const normalizedFilename = normalizePath(filename);
+  // Convert to and assume posix going forward
+  const normalizedFilename = normalizeWindowsPathSeparator ? normalizeWindowsPath(filename) : filename;
+
+  // eslint-disable-next-line prefer-const
+  let { root, dir, base: basename, ext } = posix.parse(normalizedFilename);
+
+  const base = (require && require.main && require.main.filename && dir) || global.process.cwd();
 
   // We could use optional chaining here but webpack does like that mixed with require
-  const base = normalizePath(
-    `${(require && require.main && require.main.filename && dirname(require.main.filename)) || global.process.cwd()}/`,
-  );
+  const normalizedBase = normalizeWindowsPathSeparator ? normalizeWindowsPath(`${base}/`) : base;
 
   // It's specifically a module
-  const file = basename(normalizedFilename, '.js');
```

Comment:
When I first profiled this, the regexp that runs in basename was dominating the profiles. Turns out that both basename and dirname on L26 execute the exact same regexp and just return the different components. We can hoist this into a path.parse call at the top of the function and avoid this entirely